### PR TITLE
partners: add similarity search by image functionality to langchain_chroma partner package

### PR DESCRIPTION
**Pull request by mrugank-wadekar**
- **Description:** This pull request introduces two new methods to the Langchain Chroma partner package that enable similarity search based on image embeddings. These methods enhance the package's functionality by allowing users to search for images similar to a given image URI. Also introduces a notebook to demonstrate it's use.
  - **Issue:** N/A
  - **Dependencies:** None
  - **Twitter handle:** @mrugank9009

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
way too many files have been changes, probably need to rebase or something

**Comment by mrugank-wadekar** [comment]
Hi this is my first time contributing,
The first commit files were automatically changed, should I create a new PR so I will revert those auto changed files, and only push the actual code changes in it.?
Thanks!

**Comment by mrugank-wadekar** [comment]
I've modified my branch to affect only the .py files, do let me know if there's anything more I need to do.
Thanks!

**Comment by ccurme** [comment]
@tazarov would you mind taking a look at this one? Thanks!

**Comment by mrugank-wadekar** [comment]
Hi @tazarov , can i expect a merge with this one?

**Comment by mrugank-wadekar** [comment]
Would updating the branch help for CI Success? If not, would appreciate some guidance on this. Thanks!

File: libs/partners/chroma/langchain_chroma/vectorstores.py [hunk 1]
```diff
@@ -606,6 +606,94 @@ def _select_relevance_score_fn(self) -> Callable[[float], float]:
...
+            0th element in each tuple is a Langchain Document Object.
+            The page content is b64 encoded img, metadata is default or defined by user.
+
+        Raises:
+            ValueError: If the embedding function does not support image embeddings.
+        """
+        if self._embedding_function is None or not hasattr(
+            self._embedding_function, "embed_image"
+        ):
+            raise ValueError("The embedding function must support image embedding.")
```
Comment (ccurme) [inline libs/partners/chroma/langchain_chroma/vectorstores.py:681]: Consider explicitly spelling out the required name here ("embed_image")

**Review by ccurme — APPROVED** [review]
Thanks! For future reference, CI was held up by code formatting. Please see contribution guidelines [here](https://python.langchain.com/v0.2/docs/contributing/code/setup/#formatting-and-linting) for how to resolve.

Left a minor comment and would welcome any follow-up PRs. If you can put together a concise ~10 line snippet for demonstration, would be worth adding it:

- To [docs](https://python.langchain.com/v0.2/docs/integrations/vectorstores/chroma/) (by editing notebook [here](https://github.com/langchain-ai/langchain/blob/master/docs/docs/integrations/vectorstores/chroma.ipynb);
- As an integration test.

**Comment by mrugank-wadekar** [comment]
Hi thanks, for merging!
Learned a lot in my first PR into langchain.
I will add a small demonstration in the docs notebook for Chroma, in a separate PR.
Thanks!

**Pull request merged**